### ISSUE TITLE
Cut off at the first few seconds of every song

### ISSUE THREAD

**Issue by jeffjiangg** [vincentneo/LosslessSwitcher#82]
When a song start playing, there is always a few pauses (or cut offs) for 0.5-1 second then resume playing. Sometimes there are multiple pauses and electromagnetic noise. It happens to ~80% of songs and only happens within the first 0.5-2 seconds of each song. Happens to all streaming and downloaded songs. When I quit the LosslessSwitcher app, it's gone.

MacBook Air: 2020 M1 base model
OS: 13.0.1 (22A400)
LosslessSwitcher: version 1.1, build 14
Apple Music: 1.3.0.138
all up to date

Thanks for help!

**Comment by vincentneo** [comment]
This is expected due to how detection works right now.

Here's an example, where I will refer LosslessSwitcher as LS (for simplicity), of betas v1.1 behaviour, where user will play a 96kHz track.

Essentially,
1. LS is running in background.
2. User starts playback of a song.
3. LS understands that (1) song has began playback, (2) song is likely <48kHz/96kHz> sample rate
4. LS attempts to switch audio device to <48kHz/96kHz>
5. User hears a momentary cut-off, due to sudden switching
6. (Usually it ends here, but LS continues to read further updates) LS may now learn that song is actually <96kHz> sample rate.
7. LS attempts to switch the sample rate output of the audio device again.
8. User hears a momentary cut-off, again.

(all numbers enclosed in <> is an example. it can be any value, dependent on the content that you are try to playback)
In most cases, things end at (5), and at (3), LS detects the correct sample rate, but there are times that it doesn't and that is because of how the detection works.

So basically, to solve the issue of momentary cut-off, the sample rate detection regime basically falls apart, and that's why there has been no progress on this issue, unfortunately.

**Comment by jeffjiangg** [comment]
Understood. Thanks for the explanation and everything!
